script.js

```diff
@@ -1,4 +1,188 @@
+// Create a version of the Beat That dice game,
+// where players create the largest number they can by concatenating random dice roll digits.
+//
+// --- Base Requirements:
+// 1. There are 2 players and players take turns.
+// 2. When a player clicks Submit, the game rolls 2 dice and shows the dice rolls,
+//    for example 3 and 6.
+// 3. The player picks the order of the dice they want. For example,
+//    if they wanted the number 63, they would specify that the 2nd dice goes first.
+//    You can choose how the player specifies dice order.
+// 4. After both players have rolled and chosen dice order,
+//    the player with the higher combined number wins.
+
+// ========== GLOBAL VAR ==========
+var playerRolls = [];
+var scoreKeeper = [];
+var mode = "start";
+var player = 0;
+
+// ========== GAME PARTS ==========
+// ---------- DICE ROLL ----------
+function diceRoll() {
+  var diceNum = Math.floor(Math.random() * 6) + 1;
+  return diceNum;
+}
+
+// ---------- GERNERATE ROLLS ----------
+function storeNum(rollArray, player) {
+  // reset dice array
+  // rollArray = [];
+  // push dice rolls into array 2x
+  rollArray.push(diceRoll());
+  rollArray.push(diceRoll());
+  console.log("storeNum: " + rollArray);
+  mode = "sequencing";
+  console.log("mode: " + mode);
+  var rolled =
+    `Welcome Player ${player}. <br>You rolled <u>${rollArray[0]} for Dice 1</u> and <u>${rollArray[1]} for Dice 2</u>. <br>Choose which dice to go first (1 or 2). ` +
+    leaderboard(scoreKeeper);
+  return rolled;
+}
+
+// ---------- LEADERBOARD ----------
+// checks which player is leading and list them in descending order
+function leaderboard(scoreKeeper) {
+  var scoreboard = "<br><br><u>Scoreboard</u><br>";
+  // total player count = scoreKeeper.length
+  if (scoreKeeper.length == 2) {
+    var display = `<br><br>Current score:<br>Player 1: ${scoreKeeper[0]}<br>Player 2: ${scoreKeeper[1]}`;
+    if (scoreKeeper[1] > scoreKeeper[0]) {
+      display = `<br><br>Current score:<br>Player 2: ${scoreKeeper[1]}<br>Player 1: ${scoreKeeper[0]}`;
+    }
+    return display;
+  } else {
+    for (p = 0; p < scoreKeeper.length; p++) {
+      var playerCounter = p + 1;
+      scoreboard =
+        scoreboard +
+        `Player ${playerCounter}: ${scoreKeeper[playerCounter - 1]}<br>`;
+    }
+    return scoreboard;
+  }
+}
+
+// ---------- SEQUENCING ----------
+// generates 2-digit number from input sequencing
+function sequencing(seq, rollArray, player) {
+  var sequence = "";
+  var finalNum = "";
+  console.log("sequencing fn " + seq);
+  console.log(rollArray, "in sequencing fn");
+  // user chooses which dice roll to be in front
+  if (isNaN(seq) || seq == 0) {
```

Comment:
You missed out seq > 2 in this `if `condition here. I think we have not taught the `!` operator but it's really useful if you want to check whether seq is a valid number or not. So line 73 could be `if (!(seq <= 2 && seq > 0)) {  // tell player to only input 1 or 2 }`. 

In Javascript, the exclamation mark (“!”) symbol, called a “bang,” is the logical “not” operator. Placed in front of a boolean value, it will reverse the value, returning the opposite. So something that is `true` will become `false`, vice versa.